theano/sandbox/cuda/basic_ops.py

```diff
@@ -3449,7 +3449,14 @@ def c_code(self, node, name, inp, out, sub):
         fail = sub['fail']
         str = """
         {
-            if (CudaNdarray_is_c_contiguous(%(input)s)){
+            bool strides_all_positive = False;
+            for (int i = 0; i < CudaNdarray_NDIM(%(input)s); i++){
+                if (CudaNdarray_HOST_STRIDES(%(input)s)[i] < 0){
```

Comment:
This checks for smaller than 0.  It doesn't work with the name `strides_all_positive` because you set that variable to true when you find a negative stride.  To make it clearer reverse either the assignment or the condition.